test/validate/windows_validate.go

```diff
@@ -0,0 +1,228 @@
+package validate
+
+import (
+	"context"
+	"encoding/json"
+	"log"
+	"net"
+
+	k8sutils "github.com/Azure/azure-container-networking/test/internal/k8sutils"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	privilegedWindowsDaemonSetPath = "../manifests/load/privileged-daemonset-windows.yaml"
+	windowsNodeSelector            = "kubernetes.io/os=windows"
+)
+
+var (
+	hnsEndPpointCmd  = []string{"powershell", "-c", "Get-HnsEndpoint | ConvertTo-Json"}
+	azureVnetCmd     = []string{"powershell", "-c", "cat ../../k/azure-vnet.json"}
+	azureVnetIpamCmd = []string{"powershell", "-c", "cat ../../k/azure-vnet-ipam.json"}
+)
+
+type WindowsClient struct{}
+
+type WindowsValidator struct {
+	Validator
+}
+
+type HNSEndpoint struct {
+	MacAddress       string `json:"MacAddress"`
+	IPAddress        net.IP `json:"IPAddress"`
+	IPv6Address      net.IP `json:",omitempty"`
+	IsRemoteEndpoint bool   `json:",omitempty"`
+}
+
+type AzureVnet struct {
+	NetworkInfo NetworkInfo `json:"Network"`
+}
+
+type NetworkInfo struct {
+	ExternalInterfaces map[string]ExternalInterface `json:"ExternalInterfaces"`
+}
+
+type ExternalInterface struct {
+	Networks map[string]Network `json:"Networks"`
+}
+
+type Network struct {
+	Endpoints map[string]Endpoint `json:"Endpoints"`
+}
+
+type Endpoint struct {
+	IPAddresses []net.IPNet `json:"IPAddresses"`
+	IfName      string      `json:"IfName"`
+}
+
+type AzureVnetIpam struct {
+	IPAM AddressSpaces `json:"IPAM"`
+}
+
+type AddressSpaces struct {
+	AddrSpaces map[string]AddressSpace `json:"AddressSpaces"`
+}
+
+type AddressSpace struct {
+	Pools map[string]AddressPool `json:"Pools"`
+}
+
+type AddressPool struct {
+	Addresses map[string]AddressRecord `json:"Addresses"`
+}
+
+type AddressRecord struct {
+	Addr  net.IP
+	InUse bool
+}
+
+func (w *WindowsClient) CreateClient(ctx context.Context, clienset *kubernetes.Clientset, config *rest.Config, namespace, cni string, restartCase bool) IValidator {
```

Comment:
should we return a new Validator interface for windows only? IValidator can be used for Linux.